### DISCUSSION TITLE
Language support

### DISCUSSION THREAD

**Discussion by dgoryeo** [modelscope/FunASR#1735] [Q&A] [answered]
Does FunASR support transcription of Japanese language? 
Thanks

**Comment by LauraGPT** [comment, marked as answer]
Yes. Current FunASR has two practical Japanese transcription paths:

### 1. SenseVoiceSmall (CPU or GPU)

SenseVoiceSmall is the lighter 234M model and supports Japanese explicitly. Set `language="ja"` when the input is known to be Japanese, or use `"auto"` for automatic language detection.

```python
from funasr import AutoModel
from funasr.utils.postprocess_utils import rich_transcription_postprocess

model = AutoModel(
    model="iic/SenseVoiceSmall",
    vad_model="fsmn-vad",
    device="cpu",  # or "cuda"
)

result = model.generate(
    input="japanese.wav",
    language="ja",
    use_itn=True,
    batch_size_s=60,
)
print(rich_transcription_postprocess(result[0]["text"]))
```

### 2. Fun-ASR-Nano (GPU)

Fun-ASR-Nano supports Chinese, English, and Japanese and is the current flagship LLM-ASR path:

```python
from funasr import AutoModel

model = AutoModel(
    model="FunAudioLLM/Fun-ASR-Nano-2512",
    device="cuda",
)

result = model.generate(input="japanese.wav")
print(result[0]["text"])
```

Do **not** use `paraformer-zh` for Japanese; its listed language coverage is Chinese and English only.

Current source references:
- [Model language table](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/README.md#L140-L149)
- [FunASR model-selection guide](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/docs/model_selection.md#L35-L54)
- [SenseVoice language options](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/examples/industrial_data_pretraining/sense_voice/README.md#L165-L180)
- [SenseVoice `ja` language mapping](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/funasr/models/sense_voice/model.py#L731-L734)
